### ISSUE TITLE
Tooltip is showed when MenuSurface container is closed

### ISSUE THREAD

**Issue by MiquelRamis** [hperrin/svelte-material-ui#594]
When I use a MenuSurface that appears when I press a button and disappears when I press one of the buttons included in this MenuSurface, I encounter a problem if I have tooltips linked to these buttons: if I move over these buttons and click one of them before the tooltip appears, the MenuSurface disappears with the buttons, and then the tooltip appears in a corner of the page. 

Code sample:

<IconButton bind:this={opmenubutton} on:click={() => surface.setOpen(true) } size="button">
            <Icon component={Svg} viewBox="0 0 24 24">
                <path fill="green" d={ oper.icon }></path>
            </Icon>
</IconButton>
<MenuSurface bind:this={surface}  anchor={true}
            anchorCorner="BOTTOM_LEFT"
        >
            <div style="width: 40px; margin: 0.1em; display: flex; flex-direction: row; align-items: flex-end;">
                <SegmentedButton
                    segments={opers}
                    let:segment
                    singleSelect
                    bind:selected={oper}
                    key={(segment) => segment.name}
                >
                  <Wrapper>
                  <Segment {segment} on:click={() => closeToolBar(surface) }>
                      <Icon component={Svg} viewBox="0 0 24 24">
                          <path fill="currentColor" d={ segment.icon }></path>
                      </Icon>
                  </Segment>
                  <Tooltip showDelay={0} hideDelay={0} yPos={segment.tooltippos}>{segment.name}</Tooltip>
                  </Wrapper>
              </SegmentedButton>
            </div>
</MenuSurface>

**Comment by uthraa-manickam** [comment]
Yes. I am facing the same issue. When element is clicked tooltip appears and if the element is removed from the screen the tooltip stays right where it was created. Cannot seem to remove it without refreshing the window. Very annoying and high priority bug.

**Comment by pboguslawski** [comment]
Similar problem: after hovering mouse over button with tooltip...
```
<Wrapper>
	<Button
		on:click={async () => {
			await goto('/other/page', {
				replaceState: false
			});
		}}
		variant="unelevated"
	>						
		<Label>My Button</Label>
	</Button>
	<Tooltip>My Tip</Tooltip>
</Wrapper>
```
...and clicking this button without waiting for tooltip display, tooltip is displayed while new page is loading and stays displayed over new page.

**Comment by andriivariational** [comment]
Has anyone found workaround for this?

**Comment by andriivariational** [comment]
Fix that worked for me (holding reference to `Wrapper` and destroying it when component unmounts):
```svelte
<script lang="ts">
	import Tooltip, { Content, Wrapper } from '@smui/tooltip';
	import { onDestroy } from 'svelte';
	
	export let value = "";
	export let tooltipText = "";
	
	// Hold reference to Wrapper
	let tooltipRef: any;

	onDestroy(() => {
		// Destroy Wrapper on parent component unmount
		tooltipRef?.$destroy();
	});
</script>

<Wrapper bind:this={tooltipRef}>
	{value}
	<Tooltip ...>
		<Content>
			{tooltipText}
		</Content>
	</Tooltip>
</Wrapper>

**Comment by hperrin** [comment]
This should be fixed in the latest version, but...

Thank you for filing this issue!

I'm going to close this issue for two reasons:

1. v9 has just been released, and I'm going to close all previous issues to be revalidated against v9.
2. SMUI is about to be migrated to the [SciActive Forge](https://forge.sciactive.com/sciactive).

**If you feel this issue should be reopened, please validate that it still applies to the newest version of SMUI (v9), then reopen it.**

As part of the migration to SciActive Forge, this repo **will remain** on GitHub as a mirror, but certain GitHub features will be disabled. This might include GitHub Issues, and I'd like to clean up all of the existing issues to make sure only issues that still apply to v9 get migrated. Thank you so much for your participation!
